registry/remote/referrers.go

```diff
@@ -68,6 +70,46 @@ var (
 	errNoReferrerUpdate = errors.New("no referrer update")
 )
 
+const (
+	// opDeleteReferrerIndex represents the operation for deleting a
+	// referrer index.
+	opDeleteReferrerIndex = "DeleteReferrersIndex"
+)
+
+// ReferrersError records an error and the operation and the subject descriptor.
+type ReferrersError struct {
+	// Op represents the failing operation.
+	Op string
+	// Subject is the descriptor of referenced artifact.
+	Subject ocispec.Descriptor
+	// Err is the entity of referrers error.
+	Err error
+}
+
+// Error returns error msg of IgnorableError.
+func (e *ReferrersError) Error() string {
+	return e.Err.Error()
+}
+
+// Unwrap returns the inner error of IgnorableError.
+func (e *ReferrersError) Unwrap() error {
+	return errors.Unwrap(e.Err)
+}
+
+// IsIndexDelete tells if e is kind of error related to referrers
+// index deletion.
+func (e *ReferrersError) IsReferrersIndexDelete() bool {
+	if e.Op != opDeleteReferrerIndex {
+		return false
+	}
+
+	var errResp *errcode.ErrorResponse
+	if errors.As(e.Err, &errResp) {
+		return errResp.StatusCode == http.StatusMethodNotAllowed
+	}
+	return false
+}
```

Comment:
According to our previous discussions, the ORAS CLI and Notation CLI behaves differently in terms of referrers index delete failure.

In the ORAS CLI, all errors are considered as critical and cannot be ignored unless users choose to treat them as warnings via a flag.
In the Notation CLI, all kinds of referrers deletion failure are considered as warnings and can be ignored.

Reply:
Thus we might need to revert this method to 14842434e9d37c03dc004bdceb6005fd3aa9306c.

i.e.
```go
// IsIndexDelete tells if e is kind of error related to referrers
// index deletion.
func (e *ReferrersError) IsReferrersIndexDelete() bool {
	return e.Op != opDeleteReferrerIndex
}
```

Reply:
>  all errors are considered as critical and cannot be ignored unless users choose to treat them as warnings via a flag.

Actually we don't need to pick `StatusMethodNotAllowed` out. The above can be implemented based on this, 
can you help check if https://github.com/oras-project/oras/pull/921 matched your expectation?

Reply:
Just realize that this change is for notation but not oras. Will change as required.